### ISSUE TITLE
Add enum support

### ISSUE THREAD

**Issue by uchuu-me** [JsonMapper/JsonMapper#151]
Add enum support

**Comment by DannyvdSluijs** [comment]
Hi @uchuu-me thanks for the report. Enum are already supported, are you missing something in order to make it work?

**Comment by uchuu-me** [comment]
```php
enum Color: string
{
    case GREEN = 'GREEN';
    case RED = 'RED';
    case BLUE = 'BLUE';
}

class Toy
{
    /** @var Color[] */
    public array $colors = [];
}

$mapper = (new \JsonMapper\JsonMapperFactory())->bestFit();
$toy = new Toy();

$mapper->mapObject(json_decode('{ "color": ["GREEN", "BLUE"] }'), $toy);

var_dump($toy);
```

```
class Toy#3 (1) {
  public $colors =>
  array(0) {
  }
}
```

**Comment by DannyvdSluijs** [comment]
This was fixed in [2.14.4](https://github.com/JsonMapper/JsonMapper/releases/tag/2.14.4)
